tests/pipeline/test_blaze.py

```diff
@@ -0,0 +1,449 @@
+"""
+Tests for the blaze interface to the pipeline api.
+"""
+from __future__ import division
+
+from collections import OrderedDict
+from datetime import timedelta
+from unittest import TestCase
+import warnings
+
+import blaze as bz
+from datashape import dshape, var, Record
+import numpy as np
+import pandas as pd
+from pandas.util.testing import assert_frame_equal
+from toolz import keymap
+from toolz.curried import operator as op
+
+from zipline.pipeline import Pipeline, CustomFactor
+from zipline.pipeline.data import DataSet, BoundColumn
+from zipline.pipeline.engine import SimplePipelineEngine
+from zipline.pipeline.loaders.blaze import (
+    from_blaze,
+    BlazeLoader,
+    NoDeltasWarning,
+    NonNumpyField,
+    NonPipelineField,
+)
+from zipline.utils.test_utils import tmp_asset_finder
+
+
+nameof = op.attrgetter('name')
+dtypeof = op.attrgetter('dtype')
+
+
+class BlazeToPipelineTestCase(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.dates = dates = pd.date_range('2014-01-01', '2014-01-03')
+        dates = cls.dates.repeat(3)
+        cls.sids = sids = ord('A'), ord('B'), ord('C')
+        cls.df = df = pd.DataFrame({
+            'sid': sids * 3,
+            'value': (0, 1, 2, 1, 2, 3, 2, 3, 4),
+            'asof_date': dates,
+            'timestamp': dates,
+        })
+        cls.dshape = dshape("""
+        var * {
+            sid: ?int64,
+            value: ?float64,
+            asof_date: datetime,
+            timestamp: datetime
+        }
+        """)
+        cls.macro_df = df[df.sid == 65].drop('sid', axis=1)
+        dshape_ = OrderedDict(cls.dshape.measure.fields)
+        del dshape_['sid']
+        cls.macro_dshape = var * Record(dshape_)
+
+        cls.garbage_loader = BlazeLoader()
+
+    def test_tabular(self):
+        name = 'expr'
+        expr = bz.Data(self.df, name=name, dshape=self.dshape)
+        ds = from_blaze(
+            expr,
+            loader=self.garbage_loader,
+            no_deltas_rule='ignore',
+        )
+        self.assertEqual(ds.__name__, name)
+        self.assertTrue(issubclass(ds, DataSet))
+        self.assertEqual(
+            {c.name: c.dtype for c in ds._columns},
+            {'sid': np.int64, 'value': np.float64},
+        )
+
+        for field in ('timestamp', 'asof_date'):
+            with self.assertRaises(AttributeError) as e:
+                getattr(ds, field)
+            self.assertIn("'%s'" % field, str(e.exception))
+            self.assertIn("'datetime'", str(e.exception))
+
+        self.assertIs(
```

Comment:
Elsewhere you comment on these assertions that this is testing memoization.  Should we be consistent and add one here?